src/ansiblelint/yaml_utils.py

```diff
@@ -886,13 +887,58 @@ def loads(self, stream: str) -> Any:
     def dumps(self, data: Any) -> str:
         """Dump YAML document to string (including its preamble_comment)."""
         preamble_comment: Optional[str] = getattr(data, "preamble_comment", None)
+        self._prevent_wrapping_flow_style(data)
         with StringIO() as stream:
             if preamble_comment:
                 stream.write(preamble_comment)
             self.dump(data, stream)
             text = stream.getvalue()
         return self._post_process_yaml(text)
 
+    def _prevent_wrapping_flow_style(self, data: Any) -> None:
+        if not isinstance(data, (CommentedMap, CommentedSeq)):
+            return
+        for key, value, parent_path in nested_items_path(data):
+            if not isinstance(value, (CommentedMap, CommentedSeq)):
+                continue
+            fa: Format = value.fa  # pylint: disable=invalid-name
+            if fa.flow_style():
+                predicted_indent = self._predict_indent_length(parent_path, key)
+                predicted_width = len(str(value))
```

Comment:
Generally, this should be longer than the dumped width because strings will all be quoted with `'` using python's repr on dicts/lists, but many of the strings will end up without quotes when dumped.
However, if it includes any complex strings with a bunch of escaped characters, then it could be too short. In that case, switching to a block will make dumping such a complex string clearer/more distinct, so I think that is an okay trade-off.